File: src/test/java/com/flexpoker/table/command/aggregate/generic/RemovePlayerFromTableTest.java
package com.flexpoker.table.command.aggregate.generic;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.UUID;

import org.junit.jupiter.api.Test;

import com.flexpoker.exception.FlexPokerException;
import com.flexpoker.table.command.aggregate.testhelpers.TableTestUtils;
import com.flexpoker.table.command.events.PlayerRemovedEvent;

public class RemovePlayerFromTableTest {

    @Test
    void testRemovePlayerSuccess() {
        var existingPlayer = UUID.randomUUID();
        var table = TableTestUtils.createBasicTable(UUID.randomUUID(), UUID.randomUUID(), existingPlayer);
        table.removePlayer(existingPlayer);
        assertEquals(PlayerRemovedEvent.class, table.fetchNewEvents().get(table.fetchNewEvents().size() - 1).getClass());
    }

    @Test
    void testRemovingNonExistingPlayer() {
        var table = TableTestUtils.createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID());
        assertThrows(FlexPokerException.class, () -> table.removePlayer(UUID.randomUUID()));
    }

    @Test
    void testRemovingDuringAHand() {
        var existingPlayer = UUID.randomUUID();
        var table = TableTestUtils.createBasicTableAndStartHand(UUID.randomUUID(), UUID.randomUUID(), existingPlayer);
        assertThrows(FlexPokerException.class, () -> table.removePlayer(existingPlayer));
    }

}

File: src/test/kotlin/com/flexpoker/table/command/aggregate/generic/RemovePlayerFromTableTest.kt
package com.flexpoker.table.command.aggregate.generic

import com.flexpoker.table.command.aggregate.applyEvents
import com.flexpoker.table.command.aggregate.eventproducers.removePlayer
import com.flexpoker.table.command.aggregate.testhelpers.createBasicTable
import com.flexpoker.table.command.aggregate.testhelpers.createBasicTableAndStartHand
import com.flexpoker.table.command.events.PlayerRemovedEvent
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.util.UUID

class RemovePlayerFromTableTest {

    @Test
    fun testRemovePlayerSuccess() {
        val existingPlayer = UUID.randomUUID()
        val initState = createBasicTable(UUID.randomUUID(), UUID.randomUUID(), existingPlayer)
        val events = removePlayer(initState, existingPlayer)
        assertEquals(PlayerRemovedEvent::class.java, events[0].javaClass)
    }

    @Test
    fun testRemovingNonExistingPlayer() {
        val initState = createBasicTable(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID())
        assertThrows(IllegalArgumentException::class.java) { removePlayer(initState, UUID.randomUUID()) }
    }

    @Test
    fun testRemovingDuringAHand() {
        val existingPlayer = UUID.randomUUID()
        val events = createBasicTableAndStartHand(UUID.randomUUID(), UUID.randomUUID(), existingPlayer)
        val initState = applyEvents(events)
        assertThrows(IllegalArgumentException::class.java) { removePlayer(initState, existingPlayer) }
    }

}